coralproject/talk
# 'Select all user's comments' in admin #2027

## What does this PR do?

Adds #2027 - gmail-like behavior for selecting _all_ of a user's comments in the user detail view of admin. 

## How do I test this PR?

- In admin, select a user
- Click 'Select All'
- A new button appears, 'Select all X comments for USER'
![image](https://user-images.githubusercontent.com/5203478/47578125-319de100-d90e-11e8-815b-f6d9e2e67db7.png)
- Click new button, button switches to:
![image](https://user-images.githubusercontent.com/5203478/47578151-3f536680-d90e-11e8-9da0-5768def79196.png)
- Clicking accept/reject applies new status to _all_ of user's comments

I have one problem, hopefully with some input I can solve it:
- ~~I created a new set of comments to select in the gql query for the UserDetail container that only grabs comment id's and statuses. This then loads all of a user's comments to apply an update. I'm not sure however how to reset it to the first page after running the fetchMore a few times so right now this basically only works once. Perhaps my whole approach here is off.~~ Turns out it I missed default arguments for comment status.

And a couple TODOS:
- ~~I should probably make a fragment to share between the original query and load more query~~
- ~~Once 'Select all by user' is selected, Load More should be disabled until the selection is cleared. Further, once an action is clicked (approve/reject) when 'all by user' is selected, the list should be locked or replaced with a loading indicator, ideally with progress and a cancel button because this will take at least a few seconds.~~ Switched so the element switches to loading state while the batch is running.
